packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js

```diff
@@ -106,30 +106,12 @@ class CryptoWatchPriceFeed extends PriceFeedInterface {
     // historicalPricePeriods are ordered from oldest to newest.
     // This finds the first pricePeriod whose closeTime is after the provided time.
     const match = this.historicalPricePeriods.find((pricePeriod) => {
-      return time < pricePeriod.closeTime;
+      return time < pricePeriod.closeTime && time >= pricePeriod.openTime;
     });
 
-    // If there is no match, that means that the time was past the last data point.
```

Comment:
Good catch not to default to `currentPrice` which could cause even worst errors.